pkg/config/s3.go

```diff
@@ -2,10 +2,10 @@ package config
 
 // S3Config specifies the properties required to use S3 as the storage backend
 type S3Config struct {
-	Region                  string `validate:"required" envconfig:"AWS_REGION"`
-	Key                     string `envconfig:"AWS_ACCESS_KEY_ID"`
-	Secret                  string `envconfig:"AWS_SECRET_ACCESS_KEY"`
-	Token                   string `envconfig:"AWS_SESSION_TOKEN"`
-	Bucket                  string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
-	UseDefaultConfiguration bool   `envconfig:"AWS_USE_DEFAULT_CONFIGURATION"`
+	Region              string `validate:"required" envconfig:"AWS_REGION"`
+	Key                 string `envconfig:"AWS_ACCESS_KEY_ID"`
+	Secret              string `envconfig:"AWS_SECRET_ACCESS_KEY"`
+	Token               string `envconfig:"AWS_SESSION_TOKEN"`
+	Bucket              string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
+	CredentialsEndpoint string `envconfig:"AWS_CREDENTIALS_ENDPOINT"`
```

Comment:
@manugupt1 I'm concerned that this is going to be a breaking config change. Any way we can leave the old one, and document what the difference is?

Reply:
We can mark a field as deprecated then remove it one or two releases later?

Reply:
That's fine with me 👍